report/src/main/webapp/assets/js/newComparison/newComparison.js

```diff
@@ -0,0 +1,698 @@
+        if (currentColor !== previousColor) {
+          if (encounters > 0) {
+            simplifiedLine += ',' + encounters + '|';
+          }
+          simplifiedLine += currentColor;
+          previousColor = currentColor;
+          encounters ++;
+        } else {
+          encounters ++;
+        }
+      }
+
+      if (encounters > 0) {
+        simplifiedLine += ',' + encounters;
+      }
+
+      if (previousLine) {
+        if (previousLine === simplifiedLine) {
+          if (previousElement && previousElement.type === 'space' && previousElement.value === previousLine) {
+            simplifiedList[simplifiedList.length - 1].size ++;
+            offset ++;
+          } else {
+            var newElement = {
+              type: 'space',
+              value: simplifiedLine,
+              size: 2,
+              offset: offset
+            };
+            simplifiedList.push(newElement);
+            offset ++;
+          }
+        } else {
+          if (previousElement && previousElement.type === 'block') {
+            previousElement.size ++;
+            previousElement.content.push(simplifiedLine);
+            offset ++;
+          } else {
+            var newElement = {
+              type: 'block',
+              content: [simplifiedLine],
+              size: 1,
+              offset: offset ? offset + 1 : 0
+            };
+            simplifiedList.push(newElement);
+            offset ++;
+          }
+        }
+      }
+      previousLine = simplifiedLine;
+    }
+
+    simplifiedList.push({
+      type: 'block',
+      content: [],
+      size: 0,
+      offset: offset + 1
+    });
+
+    return simplifiedList;
+  }
+
+	loadImage(imgA).then(function (image) {
+    debug && console.log('image A loaded');
+		imgSizeA = {
+			width: image.naturalWidth,
+			height: image.naturalHeight
+		};
+		canvasA.width = contextA.width = imgSizeA.width;
+		canvasA.height = contextA.height = imgSizeA.height;
+		contextA.drawImage(image, 0, 0);
+
+    debug && console.log('imgSizeA: ', imgSizeA);
+		simpleA = simplifyImage(contextA.getImageData(0, 0, imgSizeA.width, imgSizeA.height));
+    debug && console.log('simpleA: ', simpleA);
+
+		if (imgSizeB) {
+			onImagesReady();
+		}
+	});
+
+	loadImage(imgB).then(function (image) {
+    debug && console.log('image B loaded');
+		imgSizeB = {
+			width: image.naturalWidth,
+			height: image.naturalHeight
+		};
+		canvasB.width = contextB.width = imgSizeB.width;
+		canvasB.height = contextB.height = imgSizeB.height;
+		contextB.drawImage(image, 0, 0);
+
+    debug && console.log('imgSizeB: ', imgSizeB);
+		simpleB = simplifyImage(contextB.getImageData(0, 0, imgSizeB.width, imgSizeB.height));
+    debug && console.log('simpleB: ', simpleB);
+
+		if (imgSizeA) {
+			onImagesReady();
+		}
+	});
+
+	function getColor() {
+		var color = colors[colorIndex];
+		colorIndex = (colorIndex + 1) % colors.length;
+		return color;
+	}
+
+	function processLines() {
```

Comment:
This is one monster of a function. Any way to split it into more readable chunks or simplify the logic you use?

Reply:
Can be tricky :) I'll try ;)